arangod/Replication2/ReplicatedLog/SupervisionAction.h

```diff
@@ -290,6 +284,50 @@ auto inspect(Inspector& f, UpdateLogConfigAction& x) {
   return f.object(x).fields(f.field("type", hack));
 }
 
+struct UpdateEffectiveWriteConcernAction {
+  static constexpr std::string_view name = "UpdateEffectiveWriteConcernAction";
+  size_t newEffectiveWriteConcern;
+
+  auto execute(ActionContext& ctx) const -> void {
+    ctx.modify<LogPlanSpecification>([&](LogPlanSpecification& plan) {
+      plan.participantsConfig.config.effectiveWriteConcern =
+          newEffectiveWriteConcern;
+      plan.participantsConfig.generation += 1;
+    });
+    ctx.modify<LogCurrentSupervision>(
+        [&](LogCurrentSupervision& currentSupervision) {
+          currentSupervision.assumedWriteConcern = std::min(
+              currentSupervision.assumedWriteConcern, newEffectiveWriteConcern);
```

Comment:
I was looking for that piece of logic in the supervision code and couldn't find it. While the code is correct, comparing the design document with the code, this detail is missing. Would it make sense to move the computation of the value into the supervision and only set the value here?

Reply:
@goedderz and I stubled over exactly this yesterday too; It's a bit of an unfortunate judgement call which side to put the change on; I will put the computation on the Supervision side as well, as that makes it more obvious what hapens.